Sources/Kitura/Router.swift

```diff
@@ -189,10 +189,12 @@ public class Router {
 
         if resourceExtension.isEmpty {
             fileExtension = defaultEngineFileExtension ?? ""
-            // swiftlint:disable todo
-            //TODO: Use stringByAppendingPathExtension once issue https://bugs.swift.org/browse/SR-999 is resolved
-            // swiftlint:enable todo
-            resourceWithExtension = template + "." + fileExtension
+  
+            guard let urlWithExtension = URL(string: template)?.appendingPathExtension(fileExtension) else {
+                return (fileExtension: fileExtension, resourceWithExtension: template)
```

Comment:
There is an earlier line in the function that constructs `URL(string: template)` and we should probably only do that once. Also if it fails the first time the function will early out which means this `else` clause won't ever fire. Even so, I'm not sure the return value in the `else` clause is in keeping with the function since `fileExtension` might have a value, but it won't have been added to the end of `resourceWithExtension`.

I think it would be good to store the resource URL at line 186 instead of the extension, then that can be used in this later logic which would remove the need for the second guard (since `appendingPathExtension(_:)` returns a non-optional result.

Reply:
Makes sense. I'll make the change.

Reply:
@tunniclm Let me know if 7a3c91e is what you were thinking of.